spire-flex/templates/agent-configmap.yaml

```diff
@@ -0,0 +1,21 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ $.Release.Name }}-agent-config
+  namespace: {{ $.Release.Namespace }}
+data:
+  agent_config: |
+    agent {
```

Comment:
toml vs yaml/json thing. should we start with yaml here instead? I'd much prefer yaml over toml so its yaml throughout the chart. may even make the unit tests easier. not sure.

Reply:
This is not toml.  This is HCL.  The SPIRE documentation uses extensive use of HCL, I want alignment with it.  That way we don't have our users mentally translating between what they read in the SPIRE documentation and what they might read out of a config file in K8s.
The | "pipe" indicates a multi-line string.  There is also a "pipe dash" |- option that could be used.  I believe the latter one trims before-and-after white space.

I wouldn't change this without adjusting the unit tests.

Reply:
yeah, sorry. misspoke. HCL. Same issue though.

One of the things I want most out of a chart is the ability to configure all the things. I often run into charts that are incredibly frustrating to deal with because the chart maintainers think they know exactly how a piece of software will be used at all times. My users are ok voiding warranties. While I understand and like the need for guardrails to prevent accidents, intentionally jumping over them is ok at times. I don't like being forced to fork a chart just to set a setting that the software itself supports. And waiting until upstream adds every option is often incredibly painful. So, we need a safety valve so we don't have to officially support all options while allowing users to do what they need to do at times. One way is to enable pass through of values from helm into the configfile and warn the user when they are getting off the supported path.  How do we solve this without a lot of effort on our part if not going down the yaml route?

Reply:
I intend to have all the things configurable too.  I just want to make the values.yaml cover them in ways that resemble a user interface that's not a 100% clone of the helm charts because often the helm charts have their values nested in areas which can separate concerns that are better grouped.

So, if we make values.yaml a near clone of the charts, people stop thinking in agents / servers and start thinking in pods, daemonsets, statefulsets, etc.  Not sure how far one approach or the other will go, odds are it will be something of a hybrid (or drift back to 100% aping the helm charts).

Also, I did enable this feature (mostly for testing) by default.  If you feel it is better disabled, we can chat about it in the meeting we will schedule soon.  I'm ok with either on or off, I just had it on because it seemed a better default when we get the daemonset pods launching.

In another project we didn't even try to use the liveness probes, we use the spire-agent CLI health status as and executable probe.  We have the option of adding that in as an option in the future too.